build/charts/theia/values.yaml

```diff
@@ -46,32 +47,68 @@ clickhouse:
     # power-of-two equivalents: Ei, Pi, Ti, Gi, Mi, Ki.
     size: "8Gi"
     createPersistentVolume:
-      # -- Type of PersistentVolume. Can be set to "Local" or "NFS". Please set this value
-      # to use a PersistentVolume created by Theia.
+      # -- Type of PersistentVolume. Can be set to "Local" or "NFS". Please set
+      # this value to use a PersistentVolume created by Theia.
       type: ""
       local:
-        # -- The local path. Required when type is "Local".
+        # -- The local path. Required when type is "Local". To use this Local PV
+        # for cluster, please create dir <path>/vol<i> on the Node, where i varies
+        # from 0 to shards * replicas.
         path: "/data"
-        # -- Affinity for the Local Persistent Volume. By default it requires to label the
-        # Node used to store the ClickHouse data with "antrea.io/clickhouse-data-node=".
+        # -- Affinity for the Local Persistent Volume. By default it requires to
+        # label the Node used to store the ClickHouse data with
+        # "antrea.io/clickhouse-data-node=".
         affinity: {}
       nfs:
         # -- The NFS server hostname or IP address. Required when type is "NFS".
         host: ""
         # -- The path exported on the NFS server. Required when type is "NFS".
+        # To use this NFS PV for cluster, please create dir <path>/vol<i> on the
+        # NFS server, where i varies from 0 to shards * replicas.
         path: ""
-    # -- Specification for PersistentVolumeClaim. This is ignored if createPersistentVolume.type is non-empty. 
-    # To use a custom PersistentVolume, please set 
+    # -- Specification for PersistentVolumeClaim. This is ignored if
+    # createPersistentVolume.type is non-empty. To use a custom PersistentVolume,
+    # please set 
     # storageClassName: ""
     # volumeName: "<my-pv>".
     # To dynamically provision a PersistentVolume, please set
     # storageClassName: "<my-storage-class>".
-    # Memory storage is used if both createPersistentVolume.type and persistentVolumeClaimSpec are empty.
+    # Memory storage is used if both createPersistentVolume.type and
+    # persistentVolumeClaimSpec are empty.
     persistentVolumeClaimSpec: {}
       # storageClassName: ""
       # volumeName: ""
+  cluster:
```

Comment:
For the cluster, how the monitoring script works? The problem is different instances may have differernt storage volumes.

Reply:
The monitor will be deployed only on the last replica in each shard. It will check the local table on the shard and delete records when threshold is exceeded.

Reply:
Does it mean each repilca in each shard should have the same size of storage volume?

Reply:
Yes, as each replica use the same `volumeClaimTemplates`, by default they should have the same storage size.

Reply:
What happens if the pod hosting the instance with the clickhouse monitor is destroyed for any reason? I assume it's reloaded again using the template with the monitor, is that correct?

Idea for a future enhancement: add leader election to the clickhouse monitor so that only an instance in a given shard will be active at any time.

Reply:
Yes, K8s will restart the pod with the right template containing the monitor. Thanks for the suggestion of leader election! I will investigate this later.